### DISCUSSION TITLE
Handle scrolling without the need of js

### DISCUSSION THREAD

**Discussion by Revirator** [a-h/templ#1003] [Q&A] [answered]
Hey there, I was looking through the docs but could not find anything regarding my problem so here I am. I want to know if there is a way to handle the user scrolling through the page without the use of js. This is the js code I am using right now which I would like to replace if possible:

```javascript
var prevScrollY = 0;

const handleScroll = () => {
    const currentScrollY = window.scrollY;

    if (currentScrollY > prevScrollY) {
        document.getElementById("navbar").className = "hiddenNavBar";
    } else {
        document.getElementById("navbar").className = "visibleNavBar";
    }

    prevScrollY = currentScrollY;
};

window.addEventListener("scroll", handleScroll);
```

**Comment by pharrisee** [comment, marked as answer]
the templ code is transpiled to go (which in turn 'prints' HTML) before shipping to the browser, templ wouldn't have any way of influencing what happens on the browser.

**Reply by Revirator** [reply]
Ok, I see. Thanks.
